### PR TITLE
SFCC-67: Support for SFCC 6.0

### PR DESCRIPTION
**Pull request by kenredvan**
Includes new meta data, site preference, boolean that defaults to false to indicate if we are using SFCC 6.0 or something earlier (5.3).

### CODE REVIEW

File: cartridges/int_stickyio_sfra/cartridge/templates/default/checkout/shipping/shipping.isml [hunk 1]
```diff
@@ -1,33 +1,41 @@
...
                 <h2 class="card-header-custom">${Resource.msg('heading.checkout.shipping', 'checkout', null)}</h2>
             </div>
 
             <div class="shipping-error"></div>
-            <isif condition="${!!pdict.preventMultiShip === false}">
-                <isinclude template="checkout/shipping/multiShipCheckbox" />
-            </isif>
+			 <isif condition="${!!pdict.preventMultiShip === false}">
+            	<isinclude template="checkout/shipping/multiShipCheckbox" />
+			</isif>
```
Comment (asecaida) [inline cartridges/int_stickyio_sfra/cartridge/templates/default/checkout/shipping/shipping.isml:21]: Please correct spacing issues in lines 19-21?

File: cartridges/int_stickyio_sfra/cartridge/templates/default/checkout/customer/customerCard.isml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+                    </div>
+                </div>
+            </div>
+            <div class="row">
+                    <div class="col-12">
+                        <div class="mb-sm-3">
+                            <button class="btn btn-primary btn-block submit-customer-login" type="submit" name="submit" value="submit-login">
+                                ${Resource.msg('button.customer.login', 'checkout', null)}
+                            </button>
+							<isif condition="${pdict.allowGuestCheckout}">
```
Comment (asecaida) [inline cartridges/int_stickyio_sfra/cartridge/templates/default/checkout/customer/customerCard.isml:60]: Please correct spacing issues in line 60?

Comment (asecaida) [inline cartridges/int_stickyio_sfra/cartridge/templates/default/checkout/customer/customerCard.isml:60]: Please correct spacing issues in line 60?

File: cartridges/int_stickyio_sfra/cartridge/templates/default/checkout/customer/customerCard.isml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+	            <isif condition="${!pdict.customer.registeredUser}">
+	                <button class="btn btn-primary btn-block submit-customer" type="submit" name="submit" value="submit-customer">
+	                    ${Resource.msg('button.guest.customer.continue', 'checkout', null)}
+	                </button>
+	            </isif>
+	             <strong>${Resource.msg('msg.guest.customer.exists', 'checkout', null)} <button type="button" class="btn btn-link js-login-customer">${Resource.msg('action.customer.login', 'checkout', null)}</button></strong>
+	        </fieldset>
+	        <input type="hidden" name="${pdict.csrf.tokenName}" value="${pdict.csrf.token}"/>
+	    </form>
+    </isif>
```
Comment (asecaida) [inline cartridges/int_stickyio_sfra/cartridge/templates/default/checkout/customer/customerCard.isml:28]: Seems like there is an extra unneeded tab on lines 2-27.

Comment (asecaida) [inline cartridges/int_stickyio_sfra/cartridge/templates/default/checkout/customer/customerCard.isml:28]: Seems like there is an extra unneeded tab on lines 2-27.

File: cartridges/int_stickyio_sfra/cartridge/templates/default/checkout/checkout.isml [hunk 4]
```diff
@@ -62,9 +62,28 @@
 
                 <div class="alert alert-danger error-message" role="alert">
                     <p class="error-message-text"></p>
-                </div>
-
-                <!-- Step 1: Shipping -->
+                </div>	
+              			
+				<isif condition="${pdict.sfccVersion60}">
```
Comment (asecaida) [inline cartridges/int_stickyio_sfra/cartridge/templates/default/checkout/checkout.isml:67]: Please correct spacing issues in line 67?

Comment (asecaida) [inline cartridges/int_stickyio_sfra/cartridge/templates/default/checkout/checkout.isml:67]: Please correct spacing issues in line 67?

File: cartridges/int_stickyio_sfra/cartridge/controllers/CheckoutServices.js [hunk 5]
```diff
@@ -314,9 +323,12 @@ if (stickyioEnabled) {
...
             }
-
-            hooksHelper('app.customer.subscription', 'subscribeTo', [paymentForm.subscribe.checked, paymentForm.contactInfoFields.email.htmlValue], function () {});
-
+            
+            if (!sfccVersion60) {
+            	hooksHelper('app.customer.subscription', 'subscribeTo', [paymentForm.subscribe.checked, paymentForm.contactInfoFields.email.htmlValue], function () {});
+			} else {	
+            	hooksHelper('app.customer.subscription', 'subscribeTo', [paymentForm.subscribe.checked, currentBasket.customerEmail], function () {});
+			}
```
Comment (asecaida) [inline cartridges/int_stickyio_sfra/cartridge/controllers/CheckoutServices.js:331]: Please correct spacing issues in lines 328-331?

Comment (asecaida) [inline cartridges/int_stickyio_sfra/cartridge/controllers/CheckoutServices.js:331]: Please correct spacing issues in lines 328-331?

File: cartridges/int_stickyio_sfra/cartridge/controllers/CheckoutServices.js [hunk 6]
```diff
@@ -191,7 +198,9 @@ if (stickyioEnabled) {
                     currentBasket.setCustomerEmail(req.currentCustomer.profile.email);
                 } else {
                     billingAddress.setPhone(billingData.phone.value);
-                    currentBasket.setCustomerEmail(billingData.email.value);
+                   	if (!sfccVersion60) {
+       					currentBasket.setCustomerEmail(billingData.email.value);
+       				}
```
Comment (asecaida) [inline cartridges/int_stickyio_sfra/cartridge/controllers/CheckoutServices.js:203]: Please correct spacing issues in lines 201-203?

Comment (asecaida) [inline cartridges/int_stickyio_sfra/cartridge/controllers/CheckoutServices.js:203]: Please correct spacing issues in lines 201-203?

File: cartridges/int_stickyio_sfra/cartridge/controllers/CheckoutServices.js [hunk 7]
```diff
@@ -60,8 +63,12 @@ if (stickyioEnabled) {
...
             if (Object.keys(contactInfoFormErrors).length) {
                 formFieldErrors.push(contactInfoFormErrors);
-            } else {
-                viewData.email = { value: paymentForm.contactInfoFields.email.value };
+            } else {           
+            	if (!sfccVersion60) {
+             		viewData.email = { value: paymentForm.contactInfoFields.email.value };
+				} else {
+                	viewData.email = { value: currentBasket.customerEmail};
+            	}
```
Comment (asecaida) [inline cartridges/int_stickyio_sfra/cartridge/controllers/CheckoutServices.js:71]: Please correct spacing issues in lines 67-71?

Comment (asecaida) [inline cartridges/int_stickyio_sfra/cartridge/controllers/CheckoutServices.js:71]: Please correct spacing issues in lines 67-71?

File: cartridges/int_stickyio_sfra/cartridge/controllers/CheckoutServices.js [hunk 8]
```diff
@@ -23,18 +23,21 @@ if (stickyioEnabled) {
...
-
+    		var currentBasket = BasketMgr.getCurrentBasket();
             var stickyioOrder = stickyio.hasSubscriptionProducts();
 
             if (!stickyioOrder) { return next(); }
 
             var viewData = {};
             var paymentForm = server.forms.getForm('billing');
 
+			var sfccVersion60 = require('dw/system/Site').getCurrent().getCustomPreferenceValue('stickyioSFCCVersion60');
```
Comment (asecaida) [inline cartridges/int_stickyio_sfra/cartridge/controllers/CheckoutServices.js:39]: Please correct spacing issues in line 39?

Comment (asecaida) [inline cartridges/int_stickyio_sfra/cartridge/controllers/CheckoutServices.js:39]: Please correct spacing issues in line 39?

File: cartridges/int_stickyio_sfra/cartridge/controllers/CheckoutServices.js [hunk 9]
```diff
@@ -23,18 +23,21 @@ if (stickyioEnabled) {
         server.middleware.https,
         csrfProtection.validateAjaxRequest,
         function (req, res, next) {
+        	var BasketMgr = require('dw/order/BasketMgr');
             var PaymentManager = require('dw/order/PaymentMgr');
             var HookManager = require('dw/system/HookMgr');
             var Resource = require('dw/web/Resource');
             var COHelpers = require('*/cartridge/scripts/checkout/checkoutHelpers');
-
+    		var currentBasket = BasketMgr.getCurrentBasket();
```
Comment (asecaida) [inline cartridges/int_stickyio_sfra/cartridge/controllers/CheckoutServices.js:31]: Please correct spacing issues in line 31?

Comment (asecaida) [inline cartridges/int_stickyio_sfra/cartridge/controllers/CheckoutServices.js:31]: Please correct spacing issues in line 31?

File: cartridges/int_stickyio_sfra/cartridge/controllers/Checkout.js [hunk 10]
```diff
@@ -6,42 +6,69 @@ server.extend(Checkout);
...
+                // for the moment, force a consumer to re-enter any saved payment information
+                // eventually, we will move to accepting existing tokens and pass those along
+                checkoutView.customer.payment = null;
+                checkoutView.customer.customerPaymentInstruments = [];
                 checkoutView.stickyioForceRegisteredCheckout = stickyioForceRegisteredCheckout;
                 checkoutView.stickyioOrder = true;
+                
+                
+            } else {
+            	checkoutView.allowGuestCheckout = true;
```
Comment (asecaida) [inline cartridges/int_stickyio_sfra/cartridge/controllers/Checkout.js:56]: Please correct spacing issues in line 56?

Comment (asecaida) [inline cartridges/int_stickyio_sfra/cartridge/controllers/Checkout.js:56]: Please correct spacing issues in line 56?

File: cartridges/int_stickyio_sfra/cartridge/controllers/Checkout.js [hunk 11]
```diff
@@ -6,42 +6,69 @@ server.extend(Checkout);
...
+
+			
             var checkoutView = res.getViewData();
+            checkoutView.sfccVersion60 = sfccVersion60;
             if (stickyio.hasSubscriptionProducts() !== false) {
+            	if (stickyioForceRegisteredCheckout) {
+            		checkoutView.allowGuestCheckout = false;
+            	} else {
+            		checkoutView.allowGuestCheckout = true;
+            	}
```
Comment (asecaida) [inline cartridges/int_stickyio_sfra/cartridge/controllers/Checkout.js:45]: Please correct spacing issues in lines 41-45?

Comment (asecaida) [inline cartridges/int_stickyio_sfra/cartridge/controllers/Checkout.js:45]: Please correct spacing issues in lines 41-45?

**Review by asecaida — CHANGES_REQUESTED** [review]

File: cartridges/int_stickyio_sfra/cartridge/controllers/CheckoutServices.js (hunk 5 above)
Comment (kenredvan) [reply cartridges/int_stickyio_sfra/cartridge/controllers/CheckoutServices.js:331]: fixed

Comment (kenredvan) [reply cartridges/int_stickyio_sfra/cartridge/controllers/CheckoutServices.js:331]: fixed

File: cartridges/int_stickyio_sfra/cartridge/templates/default/checkout/customer/customerCard.isml (hunk 3 above)
Comment (kenredvan) [reply cartridges/int_stickyio_sfra/cartridge/templates/default/checkout/customer/customerCard.isml:28]: fixed

Comment (kenredvan) [reply cartridges/int_stickyio_sfra/cartridge/templates/default/checkout/customer/customerCard.isml:28]: fixed

File: cartridges/int_stickyio_sfra/cartridge/templates/default/checkout/shipping/shipping.isml (hunk 1 above)
Comment (kenredvan) [reply cartridges/int_stickyio_sfra/cartridge/templates/default/checkout/shipping/shipping.isml]: fixed

File: cartridges/int_stickyio_sfra/cartridge/templates/default/checkout/customer/customerCard.isml (hunk 2 above)
Comment (kenredvan) [reply cartridges/int_stickyio_sfra/cartridge/templates/default/checkout/customer/customerCard.isml]: fixed

Comment (kenredvan) [reply cartridges/int_stickyio_sfra/cartridge/templates/default/checkout/customer/customerCard.isml]: fixed

File: cartridges/int_stickyio_sfra/cartridge/controllers/CheckoutServices.js (hunk 9 above)
Comment (kenredvan) [reply cartridges/int_stickyio_sfra/cartridge/controllers/CheckoutServices.js]: fixed

File: cartridges/int_stickyio_sfra/cartridge/controllers/Checkout.js (hunk 10 above)
Comment (kenredvan) [reply cartridges/int_stickyio_sfra/cartridge/controllers/Checkout.js]: fixed

Comment (kenredvan) [reply cartridges/int_stickyio_sfra/cartridge/controllers/Checkout.js]: fixed

File: cartridges/int_stickyio_sfra/cartridge/controllers/Checkout.js (hunk 11 above)
Comment (kenredvan) [reply cartridges/int_stickyio_sfra/cartridge/controllers/Checkout.js]: fixed

Comment (kenredvan) [reply cartridges/int_stickyio_sfra/cartridge/controllers/Checkout.js]: fixed

File: cartridges/int_stickyio_sfra/cartridge/controllers/CheckoutServices.js [hunk 12]
```diff
@@ -23,18 +23,21 @@ if (stickyioEnabled) {
         server.middleware.https,
         csrfProtection.validateAjaxRequest,
         function (req, res, next) {
+        	var BasketMgr = require('dw/order/BasketMgr');
```
Comment (asecaida) [inline cartridges/int_stickyio_sfra/cartridge/controllers/CheckoutServices.js:26]: Please correct spacing issues in line 26?

File: cartridges/int_stickyio_sfra/cartridge/controllers/CheckoutServices.js (hunk 5 above)
Comment (asecaida) [reply cartridges/int_stickyio_sfra/cartridge/controllers/CheckoutServices.js:331]: Please correct spacing issues in line 331?

File: cartridges/int_stickyio_sfra/cartridge/templates/default/checkout/shipping/shipping.isml [hunk 13]
```diff
@@ -1,9 +1,16 @@
-<div class="shipping-section">
+<isif condition="${pdict.sfccVersion60}">
+	<div class="card ghost shipping">
+	    <div class="card-header">
+	        <h2 class="card-header-custom">${Resource.msg('heading.checkout.shipping', 'checkout', null)}</h2>
+	    </div>
+	</div>
```
Comment (asecaida) [inline cartridges/int_stickyio_sfra/cartridge/templates/default/checkout/shipping/shipping.isml:6]: Seems like there is an extra unneeded tab on lines 2-6.

File: cartridges/int_stickyio_sfra/cartridge/templates/default/checkout/shipping/shipping.isml (hunk 1 above)
[the remark this answers is not in the document]
Comment (kenredvan) [reply cartridges/int_stickyio_sfra/cartridge/templates/default/checkout/shipping/shipping.isml]: fixed

File: cartridges/int_stickyio_sfra/cartridge/controllers/CheckoutServices.js (hunk 12 above)
Comment (kenredvan) [reply cartridges/int_stickyio_sfra/cartridge/controllers/CheckoutServices.js]: fixed

File: cartridges/int_stickyio_sfra/cartridge/templates/default/checkout/customer/customerCard.isml [hunk 14]
```diff
@@ -0,0 +1,72 @@
+<div class="card-body">
+    <isif condition="${pdict.allowGuestCheckout}">
+        <form autocomplete="on" id="guest-customer" class="guest-form" method="POST" action="${URLUtils.url('CheckoutServices-SubmitCustomer')}"
+            <isprint value=${pdict.forms.guestCustomerForm.attributes} encoding="off" /> novalidate>
+            <fieldset class="customer-information-block">
+                <div class="row">
+                    <div class="col-sm-12">
+                        <div class="form-group">
+                            <label class="form-control-label">
+		                        ${Resource.msg('field.customer.email','checkout',null)}
```
[the remark this answers is not in the document]
Comment (kenredvan) [reply cartridges/int_stickyio_sfra/cartridge/templates/default/checkout/customer/customerCard.isml]: fixed

File: cartridges/int_stickyio_sfra/cartridge/templates/default/checkout/shipping/shipping.isml (hunk 13 above)
Comment (kenredvan) [reply cartridges/int_stickyio_sfra/cartridge/templates/default/checkout/shipping/shipping.isml]: fixed

File: cartridges/int_stickyio_sfra/cartridge/templates/default/checkout/shipping/shipping.isml [hunk 15]
```diff
@@ -15,19 +22,20 @@
...
+	                    <div class="card-header">
+	                        <isinclude template="checkout/productCard/productShippingCard" />
+	                    </div>
+	
+	                    <isinclude template="checkout/shipping/shipmentCard" />
+	                </div>
+	            </isloop>
+	        </isloop>
+	    </div>
+	</isif>
```
[the remark this answers is not in the document]
Comment (kenredvan) [reply cartridges/int_stickyio_sfra/cartridge/templates/default/checkout/shipping/shipping.isml]: fixed

**Review by asecaida — APPROVED** [review]

**Pull request merged**